### PR TITLE
Create Register Defensive Domains Runbook

### PR DESCRIPTION
**Pull request by AntonyBishop**
This PR created a runbook for Registering Defensive Domains

Connects to https://github.com/ministryofjustice/operations-engineering/issues/4820

### CODE REVIEW

File: source/documentation/dns/register_defensive_domain.html.md.erb
Comment (jasonBirchall) [inline source/documentation/dns/register_defensive_domain.html.md.erb:10]: This hits me slightly wrong. Can we change it?
Perhaps:
```suggestion
This document explains the process of registering a defensive domain at the Ministry of Justice.
```

File: source/documentation/dns/register_defensive_domain.html.md.erb
Comment (jasonBirchall) [inline source/documentation/dns/register_defensive_domain.html.md.erb:14]:
```suggestion
The [Defensive Domain Registrations policy](https://security-guidance.service.justice.gov.uk/defensive-domain-registration/#defensive-domain-registrations) is maintained by the Security team. Operations Engineering is not involved in deciding which domains require defensive registration or determining the criteria for such registrations. Our role is to implement the policy once a justified requirement is presented.
```

Comment (jasonBirchall) [reply source/documentation/dns/register_defensive_domain.html.md.erb:14]: How do we get in touch with security? Who are security? What the hell is a defensive domain? :)

File: source/documentation/dns/register_defensive_domain.html.md.erb
Comment (jasonBirchall) [inline source/documentation/dns/register_defensive_domain.html.md.erb:19]: How do we get approval?

Comment (AntonyBishop) [reply source/documentation/dns/register_defensive_domain.html.md.erb:17]: I'll add some words.

Comment (jasonBirchall) [reply source/documentation/dns/register_defensive_domain.html.md.erb]: Perhaps add a "Before proceeding with the registration, ensure you have:" Just above this list.

So, like:
## Pre-requisites

Before proceeding with the registration, ensure you have:

- blah
- blah

Comment (AntonyBishop) [reply source/documentation/dns/register_defensive_domain.html.md.erb]: Wording update to clarify approval will be via email.

File: source/documentation/dns/register_defensive_domain.html.md.erb
Comment (jasonBirchall) [inline source/documentation/dns/register_defensive_domain.html.md.erb:18]:
```suggestion
## Pre-requisites

Before proceeding with the registration, ensure you have:

- A list of domains requiring defensive registration
- A copy of the approval email from the Security Team
- Access to MoJDSD AWS Route53
```

File: source/documentation/dns/register_defensive_domain.html.md.erb
Comment (jasonBirchall) [inline source/documentation/dns/register_defensive_domain.html.md.erb:79]:
```suggestion
## Add Hosted Zones to the DNS Repository

When a domain is registered, a Hosted Zone is automatically created in AWS Route 53. The steps below outline how to add this Hosted Zone for management in the [DNS repository](https://github.com/ministryofjustice/dns), along with configuring the standard DNS records required for all defensive domains.

1. **Add the Hosted Zone YAML File**  
   Create a new `.yaml` file in the `hostedzones` directory following the standard process.

2. **Configure Standard DNS Records**  
   Add the following configuration to include the standard defensive domain records:

   ```yaml
   ---
   '':
     - ttl: 300
       type: CAA
       values:
         - flags: 0
           tag: iodef
           value: mailto:certificates@digital.justice.gov.uk
         - flags: 0
           tag: issue
           value: ;
     - ttl: 300
       type: MX
       value:
         exchange: .
         preference: 0
     - ttl: 172800
       type: NS
       values:
         - ns-xxxx.awsdns-xx.org.
         - ns-xxxx.awsdns-xx.co.uk.
         - ns-xxx.awsdns-xx.com.
         - ns-xxx.awsdns-xx.net.
     - ttl: 300
       type: TXT
       value: v=spf1 -all
   '*._domainkey':
     ttl: 300
     type: TXT
     value: v=DKIM1\; p=
   _dmarc:
     ttl: 300
     type: TXT
     value: v=DMARC1\;p=reject\;sp=reject\;rua=mailto:dmarc-rua@dmarc.service.gov.uk\;
   ```

   **Note**: Update the `NS` records with the actual values generated by Route 53 during the domain registration process.

3. **Submit a Pull Request**  
   Raise a pull request for the changes following the usual process.

4. **Repeat the Process for All Registered Domains**  
   Complete the above steps for each domain that has been registered.

5. **Notify the Requester and Security Team**  
   Inform both the requester and the Security team once the process is completed for all domains.
```

Comment (jasonBirchall) [inline source/documentation/dns/register_defensive_domain.html.md.erb]: Argh, the suggest function is no good for this one. I cleaned it up a little and wrapped the codeblock in yaml. I think this reads better.
```markdown
Sure, here’s the complete version wrapped in a markdown code block:

```markdown
## Add Hosted Zones to the DNS Repository

When a domain is registered, a Hosted Zone is automatically created in AWS Route 53. The steps below outline how to add this Hosted Zone for management in the [DNS repository](https://github.com/ministryofjustice/dns), along with configuring the standard DNS records required for all defensive domains.

1. **Add the Hosted Zone YAML File**  
   Create a new `.yaml` file in the `hostedzones` directory following the standard process.

2. **Configure Standard DNS Records**  
   Add the following configuration to include the standard defensive domain records:

   ```yaml
   ---
   '':
     - ttl: 300
       type: CAA
       values:
         - flags: 0
           tag: iodef
           value: mailto:certificates@digital.justice.gov.uk
         - flags: 0
           tag: issue
           value: ;
     - ttl: 300
       type: MX
       value:
         exchange: .
         preference: 0
     - ttl: 172800
       type: NS
       values:
         - ns-xxxx.awsdns-xx.org.
         - ns-xxxx.awsdns-xx.co.uk.
         - ns-xxx.awsdns-xx.com.
         - ns-xxx.awsdns-xx.net.
     - ttl: 300
       type: TXT
       value: v=spf1 -all
   '*._domainkey':
     ttl: 300
     type: TXT
     value: v=DKIM1\; p=
   _dmarc:
     ttl: 300
     type: TXT
     value: v=DMARC1\;p=reject\;sp=reject\;rua=mailto:dmarc-rua@dmarc.service.gov.uk\;
   /```

   **Note**: Update the `NS` records with the actual values generated by Route 53 during the domain registration process.

3. **Submit a Pull Request**  
   Raise a pull request for the changes following the usual process.

4. **Repeat the Process for All Registered Domains**  
   Complete the above steps for each domain that has been registered.

5. **Notify the Requester and Security Team**  
   Inform both the requester and the Security team once the process is completed for all domains.
```

Comment (jasonBirchall) [reply source/documentation/dns/register_defensive_domain.html.md.erb]: You'll have to remove the " /" just before the closing yaml codeblock.

**Review by jasonBirchall — APPROVED** [review]

**Pull request merged**
